src/Views/BaseComponent.elm

```diff
@@ -0,0 +1,144 @@
+module Views.BaseComponent exposing (Config, view)
+
+import Autocomplete exposing (Autocomplete)
+import Data.AutocompleteSelector as AutocompleteSelector
+import Data.Country as Country exposing (Country)
+import Data.Impact exposing (Impacts)
+import Data.Impact.Definition exposing (Definition, Definitions)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Views.Format as Format
+import Views.Icon as Icon
+
+
+type alias BaseComponent component quantity =
+    { component : component
+    , quantity : quantity
+    , country : Maybe Country
+    }
```

Comment:
Hopefully it's clear enough that here the `BaseComponent` represents either
- a `MaterialInput`
- a `RecipeIngredient`

In both cases there is a component (`Material` or `Ingredient`), and a quantity (`Mass` or `Split`) and a country.

Reply:
I quite like "Element", I'll give it a go and see how it unfolds, thanks for the suggestions.